contracts/Token.sol

```diff
@@ -100,7 +99,8 @@ contract ShieldToken is OwnableUpgradeable, ERC20PausableUpgradeable {
         super._mint(account, amount);
     }
 
-    function addFrozenWallet(address wallet, uint totalAmount, uint monthlyAmount, uint initialAmount, uint afterDays, uint monthDelay) internal {
+    function addFrozenWallet(address wallet, uint totalAmount, uint monthlyAmount, uint initialAmount, uint lockDaysPeriod) internal {
+        // TODO: what if `wallet` key is not in `frozenWallets`?
```

Comment:
If it is not in froser: we will add it below.
it means that if the account is already in "frozen" list - so we will simply add extra tokens to this account.

But it's a good catch. I will double check this logic

Reply:
p.s.
 it's easier to write such things in github with comments. It's easier to notice them here, you can reply and decide what to do next.

Reply:
added check for this case in 2774e4b3dc6dc79119597f3ba53580bbf01099e5
function addFrozenWallet will revert if address is already frozen